backend/dal/Helpers/Extensions/PropertyExtensions.cs

```diff
@@ -159,6 +157,18 @@ private static IQueryable<Entity.Views.Property> GenerateCommonQuery(this PimsCo
                             && property.ProjectNumbers.Contains(project.ProjectNumber)));
             }
 
+            if (filter.InSurplusPropertyList.HasValue && filter.InSurplusPropertyList.Value)
+            {
+                var statuses = context.Workflows.Where(w => w.Code == "SPL")
+                    .SelectMany(w => w.Status).Where(x => !x.Status.IsTerminal)
+                    .Select(x => x.StatusId).Distinct().ToArray();
+
+                query = query.Where(property =>
```

Comment:
This is a very expensive query.   If when we search for ERP or SPL I think we should invert the original query.  Start with projects and return properties form that list.  Then we can apply the other filter values to the properties.

Reply:
Ahh ok, I'll switch the ERP one as well